acisfp_check/acisfp_check.py

```diff
@@ -492,6 +539,35 @@ def search_obsids_for_viols(self, limit_name, limit, observations, temp, times,
         # Finished - return the violations list
         return viols_list
 
+    def write_temps(self, outdir, times, temps):
+        """
+        Write the states record array to the file "temperatures.dat".
+
+        Parameters
+        ----------
+        outdir : string
+            The directory the file will be written to.
+        times : NumPy array
+            Times in seconds from the start of the mission
+        temps : NumPy array
+            Temperatures in Celsius
+        """
+        outfile = os.path.join(outdir, 'temperatures.dat')
+        mylog.info('Writing temperatures to %s' % outfile)
+        T = temps[self.name]
+        e = self.predict_model.comp['earthheat__fptemp'].dvals
+        temp_recs = [(times[i], DateTime(times[i]).date, T[i], e[i])
```

Comment:
It's not a hard rule, but I've pretty much moved away from recarray to either np structured array or astropy Table.

I think there are actual performance / memory / readability benefits to using Table here (not that performance really matters in practice here).  (See http://cxc.harvard.edu/mta/ASPECT/tool_doc/pydocs/Chandra.Time.html#fast-conversion-functions for `secs2date`).
```
temp_table = Table([times, secs2date(times), T, e], names=[...], copy=False)
temp_table[self.msid].format = '.2f'
..
temp_table.write(out, format='ascii', overwrite=True)
```

Reply:
I’m happy to change it—I will note that this is almost an identical copy of write_temps in acis_thermal_check aside from the extra column, so this is not new code.

Reply:
@taldcroft I should note that the version of AstroPy on flight ska2 is too old for this code--`Table.write` doesn't recognize the `overwrite` keyword and there are some issues with the format strings.